pkg/controllers/allocation/binpacking/packer.go

```diff
@@ -199,6 +213,14 @@ func instanceTypeNames(instanceTypes []cloudprovider.InstanceType) []string {
 	return names
 }
 
+func flatten(pods ...[]*v1.Pod) []*v1.Pod {
```

Comment:
It seems like this is mostly (only?) used to get a count, the actually flattened list isn't needed? Maybe just create a function that just figures out the count (and doesn't allocate?)